filebeat/harvester/reader/line.go

```diff
@@ -13,7 +13,7 @@ import (
 // using the configured codec. The reader keeps track of bytes consumed
 // from raw input stream for every decoded line.
 type Line struct {
-	rawInput   io.Reader
+	rawInput   io.ReadCloser
```

Comment:
Currently that would only simplify the tests, right? So not sure if it brings a benefit to loosen the requirement.